airbyte-config/config-models/src/main/java/io/airbyte/config/helpers/StateMessageHelper.java

```diff
@@ -6,26 +6,52 @@
 
 import com.fasterxml.jackson.core.type.TypeReference;
 import com.fasterxml.jackson.databind.JsonNode;
+import com.google.common.collect.Iterables;
 import io.airbyte.commons.json.Jsons;
 import io.airbyte.config.StateType;
 import io.airbyte.config.StateWrapper;
 import io.airbyte.protocol.models.AirbyteStateMessage;
 import io.airbyte.protocol.models.AirbyteStateMessage.AirbyteStateType;
 import java.util.List;
 import java.util.Optional;
+import java.util.function.BiFunction;
 
 public class StateMessageHelper {
 
   public static class AirbyteStateMessageListTypeReference extends TypeReference<List<AirbyteStateMessage>> {}
 
+  private static final BiFunction<List<AirbyteStateMessage>, Boolean, StateWrapper> provideGlobalState = (stateMessages, useStreamCapableState) -> {
+    if (useStreamCapableState) {
+      return new StateWrapper()
+          .withStateType(StateType.GLOBAL)
+          .withGlobal(stateMessages.get(0));
+    } else {
+      return new StateWrapper()
+          .withStateType(StateType.LEGACY)
+          .withLegacyState(stateMessages.get(0).getData());
+    }
+  };
+
+  private static final BiFunction<List<AirbyteStateMessage>, Boolean, StateWrapper> provideStreamState = (stateMessages, useStreamCapableState) -> {
+    if (useStreamCapableState) {
+      return new StateWrapper()
+          .withStateType(StateType.STREAM)
+          .withStateMessages(stateMessages);
+    } else {
+      return new StateWrapper()
+          .withStateType(StateType.LEGACY)
+          .withLegacyState(Iterables.getLast(stateMessages).getData());
```

Comment:
Are we guaranteed that these messages are ordered by ascending time? Should probably add a comment somewhere noting this assumption